integration-tests/test_s3.py

```diff
@@ -7,18 +7,38 @@
 #
 
 from __future__ import unicode_literals
+import contextlib
 import io
 import os
 import subprocess
+import uuid
 
 import smart_open
 
-_S3_URL = os.environ.get('SO_S3_URL')
-assert _S3_URL is not None, 'please set the SO_S3_URL environment variable'
+_S3_BUCKET_NAME = os.environ.get('SO_BUCKET_NAME')
+assert _S3_BUCKET_NAME is not None, 'please set the SO_BUCKET_NAME environment variable'
 
+_SO_KEY = os.environ.get('SO_KEY')
+assert _SO_KEY is not None, 'please set the SO_KEY environment variable'
 
-def initialize_bucket():
-    subprocess.check_call(['aws', 's3', 'rm', '--recursive', _S3_URL])
+
+@contextlib.contextmanager
+def temporary():
+    """Yields a bucket/URL than can be used for temporary writing.
+
+    Removes all content under the URL when exiting.
+    """
+    key = '%s/%s' % (_SO_KEY, uuid.uuid4().hex)
+    url = 's3://%s/%s' % (_S3_BUCKET_NAME, key)
+    yield url
+    subprocess.check_call(['aws', 's3', 'rm', '--recursive', url])
+
+
+def _test_case(function):
+    def inner(benchmark):
+        with temporary() as (url):
+            return function(benchmark, url)
```

Comment:
You're passing a URl to the test cases, but they seem to expect a key. This is either wrong parameter passing or inconsistent naming. Please double check this.